### PR TITLE
Use cppo to split build for oml_lite.

### PR DESCRIPTION
**Pull request by rleonid**

### CODE REVIEW

**Comment by rleonid** [comment]
This fixes #157 and #137.

**Comment by rleonid** [comment]
@smondet @struktured @ihodes Please take a look if you can.

File: pkg/full/opam [hunk 1]
```diff
@@ -1,6 +1,6 @@
 opam-version: "1.2"
 name: "oml"
-version: "0.0.5"
+version: "0.0.6"
 available: [ ocaml-version >= "4.01" ]
```
Comment (smondet) [inline pkg/full/opam]: I think now it's 4.03.0 only?

```
# File "myocamlbuild.ml", line 170, characters 38-60:
# Error: Unbound value String.lowercase_ascii
# Command exited with code 2.
```

(from `opam pin add oml -k git .` on this branch)

Comment (smondet) [inline pkg/full/opam]: https://github.com/hammerlab/oml/commit/889522b243f9fd881706d8f442d0fb7fcce86de8

File: pkg/full/opam [hunk 2]
```diff
@@ -14,6 +14,6 @@ remove: ["ocamlfind" "remove" "oml"]
 depends: [
   "ocamlfind" {build}
   "lacaml" { >= "8.0.6" }
-  "lbfgs" { = "0.8.7" }
-  "ocephes" { = "0.8" }
```
Comment (smondet) [inline pkg/full/opam:18]: could these be `depopts` now?

Comment (smondet) [inline pkg/full/opam:18]: nevermind, forgot that there were 2 opam files

File: pkg/full/opam (hunk 1 above)
Comment (rleonid) [inline pkg/full/opam]: 👍

File: Makefile [hunk 3]
```diff
@@ -1,30 +1,48 @@
...
+	ocamlbuild -build-dir $(LITE_BUILD_DIR) $(CPPO_TAG) -tag 'cppo_D(OML_LITE)' -use-ocamlfind -I src/lib oml_lite.cma oml_lite.cmxa oml_lite.cmxs && \
+	mv src/lib/_tags_orig src/lib/_tags || \
+	mv src/lib/_tags_orig src/lib/_tags
 
 build: oml.cmxa
 
+clean:
+	ocamlbuild -clean
+	ocamlbuild -build-dir $(TEST_BUILD_DIR) -clean
+	ocamlbuild -build-dir $(LITE_BUILD_DIR) -clean
```
Comment (smondet) [inline Makefile:39]: `clean` should also remove the `_driver/` directory?

Comment (rleonid) [inline Makefile:39]: `_driver` directory is no more. I replaced it with `$(TEST_BUILD_DIR)` aka `_test_build`.

**Comment by smondet** [comment]
After installing I still depend on all the c-libs:

```
$ cat `ocamlfind query oml-lite`/META
description="OCaml Math Library (Lite)"
requires="lacaml lbfgs ocephes"
version="0.0.6"
directory="."
archive(byte)="oml_lite.cma"
archive(native)="oml_lite.cmxa"
```

**Comment by rleonid** [comment]
That is confusing, I wonder what is going on.

File: pkg/lite/META [hunk 4]
```diff
@@ -0,0 +1,6 @@
+description="OCaml Math Library (Lite)"
+requires="lacaml lbfgs ocephes"
```
Comment (smondet) [inline pkg/lite/META]: @rleonid  There! :)

Comment (rleonid) [inline pkg/lite/META]: What's confusing is that's not in my source.

Comment (rleonid) [inline pkg/lite/META]: Let me pull and push.

**Comment by smondet** [comment]
I managed to build & use an `oml_lite`-enabled `IOCaml_js` notebook!

LGTM!

**Pull request merged**